### PR TITLE
Misuv 6012 SCP025| Add SE, property| Consolidate View Template for Add Accounting method change for SE Business & property

### PR DESCRIPTION
**Pull request by tharunprasath**

### CODE REVIEW

File: app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting BusinessEndDate page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  def handleSubmitRequest(isAgent: Boolean, incomeSourceType: String)(implicit user: MtdItUser[_]): Future[Result] = {
+    val (postAction, backUrl, redirect) = incomeSourceType match {
```
Comment (avelx) [inline app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala:152]: * you pattern matching over value which then you are using as a parameter, which makes things to complicate to understand;
* Suggested code refactoring:
`
Step A:
  private val actionIndividual =  {
    Map(
      SelfEmployment.key -> (routes.IncomeSourcesAccountingMethodController.submit(SelfEmployment.key),
        routes.AddBusinessAddressController.show().url,
        routes.CheckBusinessDetailsController.show()),
      UkProperty.key -> (routes.IncomeSourcesAccountingMethodController.submit(UkProperty.key),
        routes.AddIncomeSourceStartDateCheckController.showUKProperty.url,
        routes.CheckUKPropertyDetailsController.show()),
      ForeignProperty.key -> (routes.IncomeSourcesAccountingMethodController.submit(ForeignProperty.key),
        routes.AddIncomeSourceStartDateCheckController.showForeignProperty.url,
        routes.ForeignPropertyCheckDetailsController.show())
    )
  }

Step B:
val (postAction, backUrl, redirect) = if (isAgent) {
  actionAgent(incomeSourceType)
} else {
  actionIndividual(incomeSourceType)
}
`

File: app/views/incomeSources/add/IncomeSourcesAccountingMethod.scala.html [hunk 2]
```diff
@@ -51,6 +52,14 @@
     </ul>
 }
 
+@hintContent = {
+    @if(incomeSourcesType == SelfEmployment.key) {
+        <br><div>@getMessage("hint")</div></br><div>@detailsDropdown(getMessage("example"), detailsContent)</div>
```
Comment (avelx) [inline app/views/incomeSources/add/IncomeSourcesAccountingMethod.scala.html:57]: I would assume that this part is not complete as you calling context resource called "example"

Comment (tharunprasath) [reply app/views/incomeSources/add/IncomeSourcesAccountingMethod.scala.html:57]: getMessage("example") appends to the prefix (s"$prefix.$key", args: _*) for incomeSources.add.FP.AccountingMethod.example. I did it correctly?

**Review by avelx — APPROVED** [review]
approved

File: app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting BusinessEndDate page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+   private def actionIndividualSubmitRequest(incomeSourceType: String): (Call, String, Call) = {
+    incomeSourceType match {
```
Comment (avelx) [inline app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala:152]: you still pattern matching over value you passing as a parameter
why ?
(routes.IncomeSourcesAccountingMethodController.submit(incomeSourceType)

Comment (tharunprasath) [reply app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala:152]: Missed updating the value param, I updated it now.

**Pull request merged**